models/classes/listener/DataAccessControlChangedListener.php

```diff
@@ -49,7 +49,11 @@ public function handleEvent(DataAccessControlChangedEvent $event): void
         /** @noinspection PhpUnhandledExceptionInspection */
         $resource = $this->getResource($event->getResourceId());
 
-        if ($resource->isClass() && !$event->isRecursive()) {
+        /**
+         * @TODO Verify if we already had a bug before when a class ACL is updated,
+         * but not marking `recursive` if it never applied
+         */
+        if ($resource->isClass() && !$event->applyToNestedResources()) {
```

Comment:
Some points:

- Shall it be also backward compatible with the `isRecursive`?
- If we verified that TODO, can we remove it from the comment?

```suggestion
        if ($resource->isClass() && !$event->applyToNestedResources() && !$event->isRecursive()) {
```

Reply:
Done: ee19b1543e01ec72d06c96992ea31148eb8cc9b2